src/DonorDashboards/RequestHandler.php

```diff
@@ -32,7 +32,7 @@ public function filterQueryVars( $vars ) {
 	 */
 	public function parseRequest( $query ) {
 
-		if ( is_admin() && array_key_exists( 'give-generate-donor-dashboard-page', $query->query_vars ) ) {
+		if ( is_admin() && current_user_can( 'manage_options' ) && array_key_exists( 'give-generate-donor-dashboard-page', $query->query_vars ) ) {
```

Comment:
Let's add `!wp_doing_ajax()` to rule out AJAX requests. I'd like this to only happen for admins requesting a page admin-side.

Reply:
Done, ref: https://github.com/impress-org/givewp/pull/5768/commits/b500e825514b3efb1bf20c68a1dcdc0791bd974d